akka-sample-fsm-scala/src/main/scala/sample/DiningHakkers.scala

```diff
@@ -113,56 +121,49 @@ class Hakker(name: String, left: ActorRef[ChopstickMessage], right: ActorRef[Cho
   //When a hakker is waiting for the last chopstick it can either obtain it
   //and start eating, or the other chopstick was busy, and the hakker goes
   //back to think about how he should obtain his chopsticks :-)
-  def waitForOtherChopstick(chopstickToWaitFor: ActorRef[ChopstickMessage],
-                            takenChopstick: ActorRef[ChopstickMessage]): Behavior[HakkerMessage] =
-
-    Behaviors.setup { ctx =>
-      val adapter = ctx.messageAdapter(ChopstickAnswerAdaptor)
-      Behaviors.receiveMessage {
-        case ChopstickAnswerAdaptor(Taken(`chopstickToWaitFor`)) =>
-          println(s"$name has picked up ${left.path.name} and ${right.path.name} and starts to eat")
-          startEating(ctx, 5.seconds)
-
-        case ChopstickAnswerAdaptor(Busy(`chopstickToWaitFor`)) =>
-          takenChopstick ! Put(adapter)
-          startThinking(ctx, 10.milliseconds)
-        case _ =>
-          Behaviors.unhandled
-      }
+  private def waitForOtherChopstick(chopstickToWaitFor: ActorRef[ChopstickMessage],
+                            takenChopstick: ActorRef[ChopstickMessage]): Behavior[HakkerMessage] = {
+    Behaviors.receiveMessage {
+      case ChopstickAnswerAdaptor(Taken(`chopstickToWaitFor`)) =>
+        ctx.log.info("{} has picked up {} and {} and starts to eat", name, left.path.name, right.path.name)
+        startEating(ctx, 5.seconds)
+
+      case ChopstickAnswerAdaptor(Busy(`chopstickToWaitFor`)) =>
+        takenChopstick ! Put(adapter)
+        startThinking(ctx, 10.milliseconds)
+      case _ =>
+        Behaviors.unhandled
```

Comment:
or use `Behaviors.receiveMessagePartial`

Reply:
Swtiched a bunch of these to partial